pandas/core/indexes/interval.py

```diff
@@ -1015,10 +1015,11 @@ def _format_with_header(self, header, **kwargs):
 
     def _format_native_types(self, na_rep='', quoting=None, **kwargs):
         """ actually format my specific types """
-        from pandas.io.formats.format import IntervalArrayFormatter
-        return IntervalArrayFormatter(values=self,
-                                      na_rep=na_rep,
-                                      justify='all').get_result()
+        from pandas.io.formats.format import ExtensionArrayFormatter
+        return ExtensionArrayFormatter(values=self,
+                                       na_rep=na_rep,
+                                       justify='all',
```

Comment:
why do we need this new arg? just change the output tests, which are incorrect

Reply:
In 0.23.4, we didn't have the leading space for indexes.

```
In [2]: df = pd.Series(1, index=pd.IntervalIndex.from_breaks([1, 2, 3, 4])).to_frame()

In [3]: df
Out[3]:
        0
(1, 2]  1
(2, 3]  1
(3, 4]  1
```

Reply:
wasn’t this the same issue you recently adjusted for DTi? this keywords just promote inconsistency

Reply:
No I don't think so.

AFAICT, this is happing because `IntervalIndex` and `Series[Interval]` are now both using `GenericArrayFormatter` to format the values. Series need a leading space, but indexes don't. So I think things should be more consistent. If you want I can remove the keyword and go back to the old implementation which just did the formatting on its own, but I suspect you don't want that :)

Reply:
ok i guess